app/models/mixins/miq_web_server_runner_mixin.rb

```diff
@@ -33,4 +33,22 @@ def start_rails_server(options)
       server.start
     end
   end
+
+  def do_heartbeat_work
+    log_long_running_requests
+  end
+
+  CHECK_LONG_RUNNING_REQUESTS_INTERVAL = 30.seconds
+  def log_long_running_requests
+    @last_checked_hung_requests ||= Time.now.utc
+    return if @last_checked_hung_requests > CHECK_LONG_RUNNING_REQUESTS_INTERVAL.ago
+
+    RequestStartedOnMiddleware.long_running_requests.each do |request, duration, thread|
+      message = "Long running http(s) request: '#{request}' handled by ##{Process.pid}:#{thread.object_id.to_s(16)}, running for #{duration.round(2)} seconds"
```

Comment:
Would it make sense to log the `thread.backtrace` here as well?  It would help debugging engineers find out where the bad parts of the codebase exist.  If this is really a hung process, multiple checks on this on the 30 second intervals would let us know if we are looping on the same section of code on each check.

Reply:
Maybe?  I don't know if it would be too noisy though.  I hear @kbrock thinks we log too much

Reply:
😆

Reply:
For the record:  We can (and should only) do this in a followup, this was just something I thought was nice about this approach that we are able to do.

> Maybe? I don't know if it would be too noisy though.

So my thought with this, and others, was:

1.  Ideally, no requests go over 10 seconds ever.
2.  When they do, and we have a reproducible environment, with zero changes (or maybe a settings tweak+restart) we can hit a slow request and git the UI workers on a regular interval with this signal (`SIGHUP` possibly or something unused by puma).

So going back to the "stacktrace" portion, this could maybe only be something that is only turned on when we are using a control signal, and the "light mode" is what is on all the time.  

This could even be scripted so that once the "bad url" is determined, it can be hit, and another thread waits 1 minute, and then starts hitting the running UI workers with this signal.  The whole script would run until some form of a result is returned from the server (502 or otherwise).

Anyway, just some ideas.

Reply:
> Ideally, no requests go over 10 seconds ever.

🤣

yeah, we can do the backtrace with some signal but we have to be careful of puma's signal handling like you mentioned.